### ISSUE TITLE
machines: Accessibility issues

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit-machines#48]
On the Machines page, there are the following a11y issues;

Details page:
- [x] The description list has children elements of the `<dl>` that are not `<dt>` or `<dd>` elements, the only two elements that it may contain (although those elements can have other elements as children)

Storage pool:
- [x] Progress bars lack accessible names. They should use aria-label/aria-labelledby/title or, in this case, aria-hidden="true" or role="presentation" as they visually present textual information immediately to the side.

Networks:
- [x] The description list has children elements of the `<dl>` that are not `<dt>` or `<dd>` elements (see above)
- [x] Autostart checkbox lacks label. The text "Run when host boots" should be either included next to the checkbox and both the text and the checkbox wrapped with `<label>` _or_ the text should be wrapped  `<label for="">` with the for pointing to the (unique) ID of the checkbox. Also: It appears as though this might not be a PF4 checkbox.

**Comment by KKoukiou** [comment]
Checked all mentioned issues here and they are resolved.
